server/lib/src/valueset/session.rs

```diff
@@ -425,7 +427,38 @@ impl ValueSetT for ValueSetSession {
                 // Retain all else
                 _ => true,
             }
-        })
+        });
+
+        // Now, assert that there are fewer or equal sessions to the limit.
+        if self.map.len() > SESSION_MAXIMUM {
+            // At this point we will force a number of sessions to be removed. This
+            // is replication safe since other replicas will also be performing
+            // the same operation on merge, since we trim by session issuance order.
+
+            // This is a "slow path". This is becase we optimise session storage
+            // based on fast session lookup, so now we need to actually create an
+            // index based on time. We need to also clone here since we need to mutate
+            // self.map which would violate mut/imut.
+
+            warn!(
```

Comment:
This warning should include the entry's ID, so it can be found while troubleshooting.

Reply:
It does, about 10 lines below :)

Reply:
Does it though? It shows the session ID that's being trimmed, not the account

Reply:
At that point in the code, we can't see the account, because we're in the valueset. :( let me check if the caller can log what it's doing ....